hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/buckets/buckets.tsx

```diff
@@ -0,0 +1,589 @@
+        label: k,
+        value: k
+      }));
+
+      this.setState({
+        loading: false,
+        totalCount,
+        volumeBucketMap,
+        volumeOptions,
+        lastUpdated: Number(moment()),
+        showPanel: false
+      }, () => {
+        if (!this.state.selectedVolumes || this.state.selectedVolumes.length === 0) {
+          // Select all volumes if it is first page load and volume search param is not specified
+          this._handleVolumeChange([allVolumesOption, ...volumeOptions], {action: 'select-option'});
+        } else {
+          // The selected volumes remain unchanged if volumes have been previously selected
+          this._handleVolumeChange(this.state.selectedVolumes, {action: 'select-option'});
+        }
+      });
+    }).catch(error => {
+      this.setState({
+        loading: false,
+        showPanel: false
+      });
+      showDataFetchError(error.toString());
+    });
+  };
+
+  componentDidMount(): void {
+    // For initial page (re)load, we get the volume from the URL search param
+    const initialVolume = this._getVolumeSearchParam();
+    if (initialVolume) {
+      const initialVolumeOption = {label: initialVolume, value: initialVolume};
+      this.setState({
+        selectedVolumes: [initialVolumeOption]
+      });
+    }
+
+    // Fetch buckets on component mount
+    this._loadData();
+    this.autoReload.startPolling();
+  }
+
+  componentWillUnmount(): void {
+    this.autoReload.stopPolling();
+    cancelSignal && cancelSignal.abort();
+  }
+
+  onShowSizeChange = (current: number, pageSize: number) => {
+    console.log(current, pageSize);
+  };
+
+  render() {
+    const {loading, totalCount, lastUpdated, selectedColumns,
+      columnOptions, volumeOptions, selectedVolumes,
+      selectedBuckets, showPanel, currentRow, selectedLimit} = this.state;
+    const paginationConfig: PaginationConfig = {
+      showTotal: (total: number, range) => `${range[0]}-${range[1]} of ${total} buckets`,
+      showSizeChanger: true,
+      onShowSizeChange: this.onShowSizeChange
+    };
+    return (
+      <div className='buckets-container'>
+        <div className='page-header'>
+          Buckets ({totalCount})
+          <div className='filter-block'>
+            <MultiSelect
+              allowSelectAll
+              isMulti
+              className='multi-select-container'
+              options={volumeOptions}
+              closeMenuOnSelect={false}
+              hideSelectedOptions={false}
+              value={selectedVolumes}
+              allOption={allVolumesOption}
+              onChange={this._handleVolumeChange}
+            />
+            Volumes
+          </div>
+          <div className='filter-block'>
+            <MultiSelect
+              allowSelectAll
+              isMulti
+              maxShowValues={3}
+              className='multi-select-container'
+              options={columnOptions}
```

Comment:
`defaultColumns` would contain the bucket column as well. Ideally we should not be able to deselect that since the page is showing bucket info - it would not make sense to be able to remove bucket name from the details.
<img width="300" alt="Screenshot 2023-09-26 at 12 07 45 PM" src="https://github.com/apache/ozone/assets/43001336/fa298f56-4106-4ac1-93d7-106e79c342a7">

Reply:
Updated similar to the volume page.